### PR TITLE
fix(date-picker): specify partial type for flatpickrOptions

### PR DESCRIPTION
**Pull request by flisijn**
Closes carbon-design-system/carbon-components-angular#2665

Add the type for the flatpickrOptions input in the Date picker component

### CODE REVIEW

**Comment by flisijn** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by flisijn** [comment]
recheck

File: src/datepicker/datepicker.component.ts [hunk 1]
```diff
@@ -231,7 +231,7 @@ export class DatePicker implements
 
 	protected _value = [];
 
-	protected _flatpickrOptions = {
+	protected _flatpickrOptions : flatpickr.Options.Options = {
```
Comment (Akshat55) [inline src/datepicker/datepicker.component.ts:234]:
```suggestion
	protected _flatpickrOptions : Options = {
```

Additionally with this change, you should be able to remove the `@ts-ignore` comments that suppressed the type errors previously!

File: src/datepicker/datepicker.component.ts [hunk 2]
```diff
@@ -202,10 +202,10 @@ export class DatePicker implements
 	@Input() plugins = [];
 
 	@Input()
-	set flatpickrOptions(options) {
+	set flatpickrOptions(options: flatpickr.Options.Options) {
 		this._flatpickrOptions = Object.assign({}, this._flatpickrOptions, options);
 	}
-	get flatpickrOptions() {
+	get flatpickrOptions(): flatpickr.Options.Options {
```
[the remark this answers is not in the document]
Comment (Akshat55) [reply src/datepicker/datepicker.component.ts:208]: You may also need to type cast the Object that is returned. You can do so by appending `as Options;` AFTER the shallow copy (`Object.assign`)

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]

**Pull request merged**